app/presenters/tree_builder.rb

```diff
@@ -291,28 +291,65 @@ def x_build_tree(options)
     [{:key => 'root', :children => child_nodes, :expand => true}]
   end
 
+  # determine if this is an ancestry node, and return the approperiate object
+  #
+  # @param object [Hash,Array,Object] object that is possibly an ancestry node
+  # @returns [Object,Nil] Returns the object of interest if this is an ancestry tree
+  #
+  # Ancestry trees are of the form:
+  #
+  #   {Object => {Object1 => {}, Object2 => {Object2a => {}}}}
+  #
+  # Since `build_tree` uses enumeration over this tree, it canverts the hash to an array [key, value]
+  # So handles bothcases
+  #
+  def x_get_ancestry_object(object)
```

Comment:
I know that you are following the convention of `x_` method names.

When I moved the first methods from controllers to separate classes I kept the `x_` in the name. I even did not know that it stands for "Xplorer" ;-) and I wanted to preserve the naming with stuff that was not moved yet.

But I don't think that we need to keep on naming new methods that way.